packages/core/src/App/Containers/Layout/header/default-header.tsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import classNames from 'classnames';
+import { getDecimalPlaces, platforms, routes } from '@deriv/shared';
+import { observer, useStore } from '@deriv/stores';
+import { MenuLinks, PlatformSwitcher } from 'App/Components/Layout/Header';
+import { AccountsInfoLoader } from 'App/Components/Layout/Header/Components/Preloader';
+import NewVersionNotification from 'App/Containers/new-version-notification.jsx';
+import RealAccountSignup from 'App/Containers/RealAccountSignup';
+import SetAccountCurrencyModal from 'App/Containers/SetAccountCurrencyModal';
+import ToggleMenuDrawer from 'App/Components/Layout/Header/toggle-menu-drawer.jsx';
+import platform_config from 'App/Constants/platform-config';
+import { useHistory } from 'react-router-dom';
+import HeaderAccountActions from './header-account-actions';
+
+const DefaultHeader = observer(() => {
+    const { client, common, notifications, traders_hub, ui } = useStore();
+    const {
+        currency,
+        is_bot_allowed,
+        is_dxtrade_allowed,
+        is_logged_in,
+        is_logging_in,
+        is_mt5_allowed,
+        is_switching,
+        is_landing_company_loaded,
+    } = client;
+    const { app_routing_history, current_language, platform } = common;
+    const { addNotificationMessage, client_notifications, removeNotificationMessage } = notifications;
+    const { setTogglePlatformType } = traders_hub;
+    const {
+        header_extension,
+        is_desktop,
+        is_mobile,
+        is_app_disabled,
+        is_route_modal_on,
+        is_trading_assessment_for_existing_user_enabled,
+    } = ui;
+
+    const history = useHistory();
+
+    const addUpdateNotification = () => addNotificationMessage(client_notifications?.new_version_available);
+    const removeUpdateNotification = React.useCallback(
+        () => removeNotificationMessage({ key: 'new_version_available' }),
+        [removeNotificationMessage]
+    );
+
+    React.useEffect(() => {
+        document.addEventListener('IgnorePWAUpdate', removeUpdateNotification);
+        return () => document.removeEventListener('IgnorePWAUpdate', removeUpdateNotification);
+    }, [removeUpdateNotification]);
+
+    const onClickDeposit = () => history.push(routes.cashier_deposit);
+
+    const filterPlatformsForClients = payload =>
+        payload.filter(config => {
+            if (config.link_to === routes.mt5) {
+                return !is_logged_in || is_mt5_allowed;
+            }
+            if (config.link_to === routes.dxtrade) {
+                return is_dxtrade_allowed;
+            }
+            if (
+                config.link_to === routes.bot ||
+                config.href === routes.binarybot ||
+                config.href === routes.smarttrader
```

Comment:
SmartTrader route check is used for returning the status of `is_bot_allowed`? I didn't understand this check.

Reply:
i too didn't understand it 😄 
i think `DefaultHeader` never gets displayed for a logged in user.. probably thats why it worked till date. 💃🏼